src/sagemaker/estimator.py

```diff
@@ -2219,7 +2219,20 @@ def _validate_and_set_debugger_configs(self):
         ):
             self.debugger_hook_config = DebuggerHookConfig(s3_output_path=self.output_path)
         elif not self.debugger_hook_config:
-            self.debugger_hook_config = None
+            self.debugger_hook_config = False
+
+        # Disable debugger if checkpointing is enabled by the customer
+        _should_disable_debugger = False
+        if self.checkpoint_s3_uri and self.checkpoint_local_path and self.debugger_hook_config:
+            if self.instance_count > 1:
+                _should_disable_debugger = True
+            if hasattr(self, "distribution") and self.distribution is not None:
```

Comment:
Why can't we just consolidate all of this in one if check?

```
if self.instance_count > 1 or (hasattr(self, "distribution") and self.distribution is not None):
    logger.info(
                "SMDebug Does Not Currently Support Distributed Training Jobs With Checkpointing Enabled"
            )
    self.debugger_hook_config = False
```

Reply:
I did this for readability.